src/components/Application/index.js

```diff
@@ -13,7 +13,9 @@ import normalizeTheme from '../../styles/helpers/normalizeTheme';
  */
 export default function Application(props) {
     const { children, className, style, locale, theme } = props;
-    const contextValue = { locale };
+
+    const context = useContext(AppContext);
+    const contextValue = locale ? { locale } : { locale: context.locale };
```

Comment:
Talk to Leo and we decide we're going to do this.
You have reversed the useLocale and AppContext changes.
If we do this it will already be:

`const { children, className, style, locale: localLocale, theme } = props;
    const locale = useLocale(localLocale);
    const contextValue = { locale };`